modules/nested-accordion/assets/js/frontend/handlers/nested-accordion.js

```diff
@@ -67,26 +67,52 @@ export default class NestedAccordion extends Base {
 	}
 
 	linkContainer( event ) {
-		const { container, index, targetContainer } = event.detail,
+		const { container, index, targetContainer, action: { type } } = event.detail,
 			view = container.view.$el,
 			id = container.model.get( 'id' ),
 			currentId = this.$element.data( 'id' );
 
 		if ( id === currentId ) {
-			const containers = view.find( `${ this.getSettings( 'selectors.accordionContentContainers' ) }, ${ this.getSettings( 'selectors.accordionContent' ) }` ),
-				accordionItems = view.find( this.getSettings( 'selectors.accordionItems' ) ),
-				isSpecificIndex = index !== undefined,
-				targetIndex = isSpecificIndex ? index + 1 : accordionItems.length - 1,
-				targetContentContainer = targetContainer !== undefined ? targetContainer.view.$el[ 0 ] : containers[ containers.length - 1 ],
-				targetAccordionItem = accordionItems[ targetIndex ] || accordionItems[ accordionItems.length - 1 ];
-
-			targetAccordionItem.appendChild( targetContentContainer );
+			const accordionItems = view.find( this.getSettings( 'selectors.accordionItems' ) );
+			let accordionItem, contentContainer;
+
+			switch ( type ) {
+				case 'create':
```

Comment:
Nice.

Reply:
Thanks :)
We can make it even better,
I tried to add the strings as consts, but it caused lighthouse tests to fail so I had to give up on that.